enclave-manager/web/src/components/enclaves/logs/LogViewer.tsx

```diff
@@ -54,9 +111,158 @@ export const LogViewer = ({
+    if (newIndex > searchMatchesIndices.length - 1) {
+      newIndex = 0;
+    }
+    if (newIndex < 0) {
+      newIndex = searchMatchesIndices.length - 1;
+    }
+    setCurrentSearchIndex(newIndex);
+    return newIndex;
+  };
+
+  useEffect(() => {
+    if (virtuosoRef?.current && currentSearchIndex !== undefined && currentSearchIndex >= 0) {
+      virtuosoRef.current.scrollToIndex(searchMatchesIndices[currentSearchIndex]);
+    }
+  }, [currentSearchIndex]);
+
+  const clearSearch = () => {
+    setRawSearchTerm("");
+    setSearch(undefined);
+    setSearchMatchesIndices([]);
+    setCurrentSearchIndex(undefined);
+  };
+
+  const parseMatchIndexRequest = (input: string) => {
+    let parsed = parseInt(input);
+    if (isNaN(parsed) || parsed < 1) return 1;
+    if (parsed > searchMatchesIndices.length) return searchMatchesIndices.length;
+    return parsed;
+  };
+
+  const highlight = (currentSearchIndex: number | undefined, thisIndex: number, searchableIndices: number[]) => {
+    return (
+      currentSearchIndex !== undefined &&
+      searchableIndices.length > 0 &&
+      searchableIndices[currentSearchIndex] === thisIndex
+    );
+  };
+
   return (
     <Flex flexDirection={"column"} gap={"32px"} h={"100%"}>
       <Flex flexDirection={"column"} position={"relative"} bg={"gray.800"} h={"100%"}>
+        <Box width={"100%"}>
+          <Flex m={4}>
+            <Flex width={"40%"}>
+              <InputGroup size="sm">
+                <Input
+                  size={"sm"}
+                  ref={searchRef}
+                  value={rawSearchTerm}
+                  onChange={handleOnChange}
+                  placeholder={"search"}
+                />
+                {rawSearchTerm && (
+                  <InputRightElement>
+                    <SmallCloseIcon onClick={clearSearch} />
+                  </InputRightElement>
+                )}
+              </InputGroup>
+            </Flex>
+            <Button size={"sm"} ml={2} onClick={priorMatch}>
+              Previous
+            </Button>
+            <Button size={"sm"} ml={2} onClick={nextMatch}>
+              Next
+            </Button>
+            {hasSearchTerm() && (
+              <Box ml={2}>
+                <Text align={"left"} color={searchMatchesIndices.length === 0 ? "red" : "kurtosisGreen.400"}>
+                  <HStack alignItems={"center"}>
+                    <>
+                      {searchMatchesIndices.length > 0 && currentSearchIndex !== undefined && (
+                        <>
+                          <Editable
+                            p={0}
+                            m={0}
+                            size={"sm"}
+                            value={`${currentSearchIndex + 1}`}
+                            onChange={(inputString) =>
+                              updateSearchIndexBounded(parseMatchIndexRequest(inputString) - 1)
+                            }
+                          >
```

Comment:
I'm not familiar with this component, but wondering if we can apply numeric constraint to it.